Source/WebCore/page/Page.cpp

```diff
@@ -5840,6 +5840,16 @@ bool Page::requiresUserGestureForVideoPlayback() const
 }
 
 #if HAVE(SUPPORT_HDR_DISPLAY)
+bool Page::drawsHDRContent() const
+{
+    bool drawsHDRContent = false;
+    forEachDocument([&] (Document& document) {
+        if (document.drawsHDRContent())
+            drawsHDRContent = true;
```

Comment:
If we could break here it would be a little faster. Maybe `forEachRenderableDocument` ?